### PR TITLE
llama3: do not quantize the embedding matrix for int8

### PR DESCRIPTION
**Pull request by Stripteaser**
int8 sets `quantize_embeds=True`, and the output projection reuses that matrix
(`output.weight = tok_embeddings.weight`), so quantising it makes the logits noisy enough
to destroy generation -- the model emits unrelated tokens instead of text. `nf4` sets it to
`False` and its output has always been coherent, which isolated the difference.

Measured on GA102 over USB4, `BEAM=2`, Llama-3 8B:

    mode   tok/s  weight BW  of peak  GB/token  VRAM
    fp16    27.9   448 GB/s     48%     16.1    16 GB
    int8    24.5   223 GB/s     24%      9.1     9 GB
    nf4     14.7    85 GB/s      9%      5.8     6 GB

int8 goes from 4.3 tok/s of garbage to 24.5 tok/s of correct text, at 40% less VRAM.

### CODE REVIEW

**Comment by geohot** [comment]
no AI
